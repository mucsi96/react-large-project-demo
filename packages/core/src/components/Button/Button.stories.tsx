import { Meta, Story } from '@storybook/react';
import React, { ReactNode } from 'react';
import { Button, ButtonProps } from './Button';

type StoryProps = ButtonProps | { children: ReactNode };

export default { title: 'Button', component: Button } as Meta;

const Template: Story<StoryProps> = (args) => <Button {...args} />;

function createStory(args: StoryProps) {
  const story = Template.bind({});
  story.args = args;
  return story;
}

export const withText = createStory({
  children: 'Hello Button',
});

export const withEmoji = createStory({
  children: '😀 😎 👍 💯',
});
