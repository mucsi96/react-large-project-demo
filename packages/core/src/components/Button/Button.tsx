import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './Button.module.scss';

export type ButtonProps = {
  disabled?: boolean;
  /**
   * Simple click handler
   */
  onClick?: () => void;

  /**
   * Is primary?
   */
  primary?: boolean;

  /**
   * default is false
   */

  secondary?: boolean;
};

export const Button: FC<ButtonProps> = ({
  children,
  onClick,
  disabled,
  primary,
  secondary,
}) => (
  <button
    type="button"
    onClick={onClick}
    className={classNames(styles.container, {
      [styles.primary]: primary,
      [styles.secondary]: secondary,
      [styles.disabled]: disabled,
    })}
    disabled={disabled}
  >
    {children}
  </button>
);
