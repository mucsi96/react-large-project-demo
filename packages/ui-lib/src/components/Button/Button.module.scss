.container {
  vertical-align: top;
  font-family: inherit;
  padding: 10px 20px;
  border: 1px solid white;
  border-radius: 10px;
}

.primary {
  background-color: chocolate;
  color: white;
}

.secondary {
  background-color: #cecece;
  color: black;
}
