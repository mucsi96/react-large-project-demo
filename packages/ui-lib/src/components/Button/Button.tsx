import classNames from "classnames";
import React, { FC } from "react";
import styles from "./Button.module.scss";

export type ButtonProps = {
  /**
   * Simple click handler
   */
  onClick?: () => void;

  /**
   * Is primary?
   */
  primary?: boolean;

  /**
   * default is false
   */
  secondary?: boolean;
};

export const Button: FC<ButtonProps> = ({
  children,
  onClick,
  primary,
  secondary,
}) => (
  <button
    type="button"
    onClick={onClick}
    className={classNames({
      [styles.primary]: primary,
      [styles.secondary]: secondary,
    })}
  >
    {children}
  </button>
);
