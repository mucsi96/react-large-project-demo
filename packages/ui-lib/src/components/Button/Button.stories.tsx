import { action } from "@storybook/addon-actions";
import React from "react";
import { Button } from "./Button";

export default { title: "Button", component: Button };

export const withText = () => (
  <Button onClick={action("clicked")}>Hello Button</Button>
);

export const withEmoji = () => (
  <Button onClick={action("clicked")}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
);
