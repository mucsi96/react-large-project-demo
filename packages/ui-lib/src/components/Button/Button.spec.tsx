import { shallow } from "enzyme";
import React from "react";
import { Button } from "./Button";

test("render", () => {
  const wrapper = shallow(<Button onClick={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
