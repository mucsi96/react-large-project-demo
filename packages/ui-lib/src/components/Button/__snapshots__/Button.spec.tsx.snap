// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
/>
`;
